src/AutoPilotPlugins/PX4/PX4PIDTuningComponent.cc

```diff
@@ -0,0 +1,113 @@
+/*=====================================================================
+
+ QGroundControl Open Source Ground Control Station
+
+ (c) 2009 - 2014 QGROUNDCONTROL PROJECT <http://www.qgroundcontrol.org>
+
+ This file is part of the QGROUNDCONTROL project
+
+ QGROUNDCONTROL is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ QGROUNDCONTROL is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with QGROUNDCONTROL. If not, see <http://www.gnu.org/licenses/>.
+
+ ======================================================================*/
+
+#include "PX4PIDTuningComponent.h"
+#include "PX4AutoPilotPlugin.h"
+#include "AirframeComponent.h"
+
+PX4PIDTuningComponent::PX4PIDTuningComponent(Vehicle* vehicle, AutoPilotPlugin* autopilot, QObject* parent)
+    : VehicleComponent(vehicle, autopilot, parent)
+    , _name("PID Tuning")
+{
+}
+
+QString PX4PIDTuningComponent::name(void) const
+{
+    return _name;
+}
+
+QString PX4PIDTuningComponent::description(void) const
+{
+    return tr("The PID Tuning Component is used to tune the PID controllers of the Vehicle.");
+}
+
+QString PX4PIDTuningComponent::iconResource(void) const
+{
+    return "/qmlimages/PIDTuningComponentIcon.png";
+}
+
+bool PX4PIDTuningComponent::requiresSetup(void) const
+{
+    return false;
+}
+
+bool PX4PIDTuningComponent::setupComplete(void) const
+{
+    return true;
+}
+
+QStringList PX4PIDTuningComponent::setupCompleteChangedTriggerList(void) const
+{
+    return QStringList();
+}
+
+QUrl PX4PIDTuningComponent::setupSource(void) const
+{
+    QString qmlFile;
+
+    switch (_vehicle->vehicleType()) {
```

Comment:
Those Vehicle::fixedWing ... things are new and the PX4TuningComponent code is older than that. Could you fix up the PX4TuningComponent code as well to use the new way to do it while you are in this space?